Community-Supported/clouddb-extractor/bigquery_extractor.py

```diff
@@ -0,0 +1,514 @@
+        blob.download_to_filename("{}.gz".format(local_filename))
+        subprocess.run(["gunzip", "{}.gz".format(local_filename)], check=True)
+
+    def export_load(
+        self, source_table, tab_ds_name, publish_mode=TSC.Server.PublishMode.CreateNew
+    ):
+        """
+        Bulk export the contents of source_table and load to a Tableau Server
+
+        source_table (string): Source table ref ("project ID.dataset ID.table ID")
+        tab_ds_name (string): Target datasource name
+        publish_mode: One of TSC.Server.[Overwrite|CreateNew] (default=CreateNew)
+        """
+        # Uses the bigquery export api to split large table to csv for load
+        source_table_ref = bq_client.get_table(source_table)
+        target_table_def = self._hyper_table_definition(source_table_ref)
+        first_chunk = True
+        for blob in self._extract_to_blobs(source_table):
+            temp_csv_filename = tempfile_name(prefix="temp", suffix=".csv")
+            self._download_blob(blob, temp_csv_filename)
+            output_hyper_files = self._csv_to_hyper_files(
+                temp_csv_filename, target_table_def
+            )
+            for path_to_database in output_hyper_files:
+                if first_chunk:
+                    self._publish_hyper_file(
+                        path_to_database, tab_ds_name, publish_mode
+                    )
+                    first_chunk = False
+                else:
+                    self._update_datasource_from_hyper_file(
+                        path_to_database=path_to_database,
+                        tab_ds_name=tab_ds_name,
+                        action="INSERT",
+                    )
+                os.remove(path_to_database)
+            os.remove(Path(temp_csv_filename))
+
+    def append_to_datasource(
+        self,
+        tab_ds_name,
+        sql_query=None,
+        source_table=None,
+        changeset_table_name="new_rows",
+    ):
+        """
+        Appends the result of sql_query to a datasource on Tableau Server
+
+        tab_ds_name (string): Target datasource name
+        sql_query (string): The query string that generates the changeset
+        source_table (string): Identifier for source table containing the changeset
+        changeset_table_name (string): The name of the table in the hyper file that
+            contains the changeset (default="new_rows")
+
+        NOTES:
+        - Specify either sql_query OR source_table, error if both specified
+        """
+
+        if not (bool(sql_query) ^ bool(source_table)):
+            raise Exception("Must specify either sql_query OR source_table")
+        if sql_query:
+            # Execute query to generate append changeset - slower than bulk extract
+            output_hyper_files = self._query_to_hyper_files(
+                sql_query, changeset_table_name
+            )
+
+            for path_to_database in output_hyper_files:
+                self._update_datasource_from_hyper_file(
```

Comment:
as mentioned above: it's a bad idea to chunk the payload into multiple Hyper files. Let's remove this loop...

`_query_to_hyper_files` only returns a single file anyway. Probably want to rename that to `_query_to_hyper_file`